temporalio/envconfig.py

```diff
@@ -0,0 +1,408 @@
+            grpc_meta=d.get("grpc_meta") or {},
+        )
+
+    def to_client_connect_config(self) -> ClientConnectConfig:
+        """Create a `ClientConnectConfig` from this profile."""
+        config: ClientConnectConfig = {}
+        if self.address:
+            config["target_host"] = self.address
+        if self.namespace:
+            config["namespace"] = self.namespace
+        if self.api_key:
+            config["api_key"] = self.api_key
+        if self.tls:
+            config["tls"] = self.tls.to_connect_tls_config()
+        if self.grpc_meta:
+            config["rpc_metadata"] = self.grpc_meta
+        return config
+
+
+@dataclass
+class ClientConfig:
+    """Client configuration loaded from TOML and environment variables.
+
+    This contains a mapping of profile names to client profiles. Use
+    `ClientConfigProfile.to_connect_config` to create a `ClientConnectConfig`
+    from a profile. See `load_profile` to load an individual profile.
+
+    .. warning::
+        Experimental API.
+    """
+
+    profiles: Mapping[str, ClientConfigProfile]
+    """Map of profile name to its corresponding ClientConfigProfile."""
+
+    @staticmethod
+    def _from_bridge_profiles(
+        bridge_profiles: Mapping[str, Mapping[str, Any]],
+    ) -> ClientConfig:
+        return ClientConfig(
+            profiles={
+                k: ClientConfigProfile.from_dict(v) for k, v in bridge_profiles.items()
+            }
+        )
+
+    @staticmethod
+    def load_profiles(
+        *,
+        disable_file: bool = False,
+        config_file_strict: bool = False,
+        env_vars: Optional[Mapping[str, str]] = None,
+    ) -> ClientConfig:
+        """Load all client profiles from default file locations and environment variables.
+
+        This does not apply environment variable overrides to the profiles, it
+        only uses an environment variable to find the default config file path
+        (`TEMPORAL_CONFIG_FILE`). To get a single profile with environment variables
+        applied, use `load_profile`.
+
+        Args:
+            disable_file: If true, file loading is disabled. Will create a default
+                configuration.
+            config_file_strict: If true, will TOML file parsing will error on
+                unrecognized keys.
+            env_vars: The environment variables to use for locating the default config
+                file. If not provided, `TEMPORAL_CONFIG_FILE` is not checked
+                and only the default path is used (./temporal/temporal.toml). To use
+                the current process's environment, `os.environ` can be passed explicitly.
+        """
+        loaded_profiles = _bridge_envconfig.load_client_config(
+            disable_file=disable_file,
+            config_file_strict=config_file_strict,
+            env_vars=env_vars,
+        )
+        return ClientConfig._from_bridge_profiles(loaded_profiles)
+
+    @staticmethod
+    def load_profiles_from_file(
```

Comment:
I wonder if we can combine all of these `load` calls into one to reduce surface area since they are not the commonly called methods. I know Go doesn't combine them because it's a bit harder for that language. I figure you have one `load` call that can accept config file override (can be path or data like cert), disable env, override env, etc.

As for the `load_profile` stuff, wouldn't that just be `load` on `ClientConfigProfile` instead of this class? Most of the same parameters should apply.

Reply:
done, some changes to the bridge to support this